### ISSUE TITLE
IP address assignment order not consistent with private_ips

### ISSUE THREAD

**Issue by JeffGiroux** [F5Networks/terraform-aws-bigip-module#24]
User needs consistent IP address assignment when there is a primary and secondary address both assigned to the NIC.

## Current behavior
The AWS resource provider object 'aws_network_interface' has a parameter for 'private_ips'. This is currently the resource type and parameter in use to build the BIG-IP NICs. However, according to AWS docs and my testing...the IP assignments can be out of order.

As a result...this code might assign primary IP as secondary and vice versa. This will have a negative affect on sending traffic to the correct IP address in the cloud provider.

```
resource "aws_network_interface" "public" {
  count             = length(compact(local.external_public_private_ip_primary)) > 0 ? length(local.external_public_subnet_id) : 0
  subnet_id         = local.external_public_subnet_id[count.index]
  security_groups   = var.external_securitygroup_ids
  private_ips       = [local.external_public_private_ip_primary[count.index], local.external_public_private_ip_secondary[count.index]]
  source_dest_check = var.external_source_dest_check
  tags = merge(local.tags, {
    Name = format("%s-%d", "BIGIP-External-Public-Interface", count.index)
    }
  )
}
```

Per AWS docs...
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/network_interface

private_ips = (Optional) List of private IPs to assign to the ENI without regard to order.

## Workaround
Two other parameters should be used instead when assigning multiple IPs to the BIG-IP. 

private_ip_list = (Optional) List of private IPs to assign to the ENI in sequential order. Requires setting private_ip_list_enabled to true.
private_ip_list_enabled = (Optional) Whether private_ip_list is allowed and controls the IPs to assign to the ENI and private_ips and private_ips_count become read-only. Default false.

Using the 'private_ip_list' and the 'private_ip_list_enabled' will ensure proper IP address order assigned to the NIC in AWS. Please make the updates.

Example fix...
```
resource "aws_network_interface" "public" {
  count                   = length(compact(local.external_public_private_ip_primary)) > 0 ? length(local.external_public_subnet_id) : 0
  subnet_id               = local.external_public_subnet_id[count.index]
  security_groups         = var.external_securitygroup_ids
  private_ip_list_enabled = true
  private_ip_list         = [local.external_public_private_ip_primary[count.index], local.external_public_private_ip_secondary[count.index]]
  source_dest_check       = var.external_source_dest_check
  tags = merge(local.tags, {
    Name = format("%s-%d", "BIGIP-External-Public-Interface", count.index)
    }
  )
}
```

**Comment by KrithikaChidambaram** [comment]
This request is now being tracked internally with ID INFRAANO-860. Thanks!

**Comment by RavinderReddyF5** [comment]
@JeffGiroux fixed in v1.1.7, please verify
